app/controllers/bus_stops_controller.rb

```diff
@@ -77,15 +79,15 @@ def update
 
   def find_stop
     @stop = BusStop.find_by hastus_id: params.require(:id)
-    @data_fields = @stop.data_fields
     unless @stop.present?
       redirect_back(fallback_location: root_path,
         notice: "Stop #{params[:id]} not found") and return
     end
+    @data_fields = @stop.data_fields
   end
 
   def stop_params
     # no attributes that people aren't supposed to be able to edit
-    params.require(:bus_stop).permit!
+    params.require(:bus_stop).permit(:completed, bus_stop_fields_attributes: [:field_name, :value])
```

Comment:
Where does this get parsed?

Reply:
Oh, I saw it on the server. It gets parsed after I click `save stop`

Reply:
What code parses the data structure, is what I mean?

Reply:
Answered https://github.com/umts/stop-project/pull/107#discussion_r201771018